src/adm/filter.rs

```diff
@@ -181,28 +187,25 @@ impl AdmFilter {
     /// Try to update the ADM filter data from the remote bucket.
     pub async fn update(&mut self, storage_client: &cloud_storage::Client) -> HandlerResult<()> {
         if let Some(bucket) = &self.source_url {
-            let adm_settings = AdmFilterSettings::from_settings_bucket(storage_client, bucket)
-                .await
-                .map_err(|e| {
-                    HandlerError::internal(&format!(
-                        "Invalid bucket data in {:?}: {:?}",
-                        self.source, e
-                    ))
-                })?;
-            for (adv, setting) in adm_settings.advertisers {
-                if setting.delete {
-                    trace!("Removing advertiser {:?}", &adv);
-                    self.filter_set.remove(&adv.to_lowercase());
-                };
-                trace!("Processing records for {:?}", &adv);
-                // DEFAULT included but sans special processing -- close enough
-                for country in &setting.include_regions {
-                    if !self.all_include_regions.contains(country) {
-                        self.all_include_regions.insert(country.clone());
+            let advertiser_filters =
+                AdmFilter::advertisers_from_settings_bucket(storage_client, bucket)
+                    .await
+                    .map_err(|e| {
+                        HandlerError::internal(&format!(
+                            "Invalid bucket data in {:?}: {:?}",
+                            self.source, e
+                        ))
+                    })?;
+            self.advertiser_filters.adm_advertisers.clear();
+            for (adv, setting) in advertiser_filters.adm_advertisers {
+                for country in setting.keys() {
+                    if !self.defaults.include_regions.contains(country) {
+                        self.defaults.include_regions.push(country.clone());
```

Comment:
wasn't sure if this was the way to go about adding include_regions

Reply:
I don't think `include_regions` belongs in `AdmDefaults`, it's a bit confusing there since it should only be generated by contile. Despite the previous `DEFAULT` key supporting an `include_regions` value (which was a bit different) we didn't really need it and especially don't with shepherd's UI and the new per region advertiser settings, so it was good that you originally got rid of it.

Let's keep it as `AdmFilter::all_include_regions` (`all_`'s helpful to distinguish it from the `include_regions` settings) and as a `HashSet`. I think you can pretty much revert to most of the previous code, and this `update()` could easily `.clear()` and rebuild it.

cc @jrconlin we can probably kill `NoTilesForCountry` as well?

Reply:
I think we need to keep `NoTilesForCountry`. We have logic still in `filter_and_process` to optionally return a 200 for tiles that are in excluded countries (since those default to return 204), and `NoTilesForCountry` allows us to return that in `error.rs`.

Reply:
I see what you're getting at: The `settings.excluded_countries_200` flag works around [what's really a bug in the client](https://github.com/mozilla-services/contile/issues/284) where it treats our 204 EMPTY response as a "noop, reverting back to the existing tile set it has", so this setting makes us return a 200 OK instead, forcing the client to use an empty tile set instead.

You're right in that we've only applied this setting to that one code path and we're still rendering 204s in other cases, with `NoTilesForCountry` covering one of them.

However there is a bug with it, it basically "short circuits" the filtering process, I've made a test on another branch which it causes to fail: https://github.com/mozilla-services/contile/compare/shepherd-output...shepherd-output-filter-bug (we're returning empty tiles there instead of 1)

I think it makes sense to consider the other 204s we're still rendering as a bug (separate from this PR) but I think the right fix is passing this setting along to all the places rendering 204s where they can take it into account.